terminus-ui/chip/src/chip.component.ts

```diff
@@ -0,0 +1,419 @@
+  public chipCollectionSelectable = true;
+
+  /** Whether the chip collection is in multi-selection mode. */
+  public chipCollectionMultiple = false;
+
+
+  /** The chip's trailing icon. */
+  @ContentChild(TsChipTrailingIconDirective, {static: false}) public trailingIcon!: TsChipTrailingIconDirective;
+
+  /** The chip's remove toggler. */
+  @ContentChild(forwardRef(() => TsChipRemoveDirective), {static: false}) public removeIcon!: TsChipRemoveDirective;
+
+  @Input()
+  public isDisabled = false;
+
+  /**
+   * Define an ID for the component
+   */
+  @Input()
+  public set id(value: string) {
+    this._id = value || this.uid;
+  }
+  public get id(): string {
+    return this._id;
+  }
+  protected _id: string = this.uid;
+
+
+  /** Whether the chip is selected. */
+  @Input()
+  public get selected(): boolean {
+    return this._selected;
+  }
+  public set selected(value: boolean) {
+    const coercedValue = coerceBooleanProperty(value);
+
+    if (coercedValue !== this._selected) {
+      this._selected = coercedValue;
+      this.dispatchSelectionChange();
+    }
+  }
+  protected _selected = false;
+
+  /** The value of the chip. Defaults to the content inside `<ts-chip>` tags. */
+  @Input()
+  public get value(): string {
+    return this._value === undefined
+      ? this.elementRef.nativeElement.textContent : this._value;
+  }
+  public set value(value: string) {
+    this._value = value;
+  }
+  protected _value = '';
+
+  /**
+   * Whether or not the chip is selectable. By default a chip is
+   * selectable, and it becomes non-selectable if its parent chip collection is
+   * not selectable.
+   */
+  @Input()
+  public get isSelectable(): boolean {
+    return this._selectable && this.chipCollectionSelectable;
+  }
+  public set isSelectable(value: boolean) {
+    this._selectable = coerceBooleanProperty(value);
+  }
+  protected _selectable = true;
+
+  /**
+   * Determines whether or not the chip displays the remove styling and emits (removed) events.
+   */
+  @Input()
+  public get isRemovable(): boolean {
+    return this._removable;
+  }
+  public set isRemovable(value: boolean) {
+    this._removable = coerceBooleanProperty(value);
+    this.changeDetectorRef.detectChanges();
+  }
+  protected _removable = true;
+
+  /** Emits when the chip is focused. */
+  public readonly onFocus = new Subject<TsChipEvent>();
+
+  /** Emits when the chip is blured. */
+  public readonly onBlur = new Subject<TsChipEvent>();
+
+  /** Emitted when the chip is selected or deselected. */
+  @Output() public readonly selectionChange: EventEmitter<TsChipSelectionChange> =
+      new EventEmitter<TsChipSelectionChange>();
+
+  /** Emitted when the chip is destroyed. */
+  @Output() public readonly destroyed: EventEmitter<TsChipEvent> = new EventEmitter<TsChipEvent>();
+
+  /** Emitted when a chip is to be removed. */
+  @Output() public readonly removed: EventEmitter<TsChipEvent> = new EventEmitter<TsChipEvent>();
```

Comment:
Funny that I used to do what you say here, but started to add typing because I saw somewhere having it and thought it's convention now to add it lol
now undo what i did...